### PR TITLE
add add_new_algorithm.md

### PR DESCRIPTION
**Pull request by cuicheng01**

### CODE REVIEW

File: docs/zh_CN/community/add_new_algorithm.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+│       ├── operators.py # 常见的数据预处理
+│       ├── randaugment.py # 随机增强预处理
+│       └── ......
+└── utils
+    └── get_image_list.py # 将文件夹中的图片文件转换为img list
+```
+
+PaddleClas 内置了大量图像操作相关模块，对于没有内置的模块可通过如下步骤添加:
+
+1. 如果只涉及单个图像的操作，在 [ppcls/data/preprocess/ops](../../../ppcls/data/preprocess/ops) 文件夹下新建文件，如果设计整个batch的图像操作，需要在 [ppcls/data/preprocess/batch_ops](../../../ppcls/data/preprocess/batch_ops) 文件夹下新建文件，如my_module.py。
```
Comment (TingquanGao) [inline docs/zh_CN/community/add_new_algorithm.md]: 设计 -> 涉及

File: docs/zh_CN/community/add_new_algorithm.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+  - MyModule:
+      args1: args1
+      args2: args2
+```
+
+<a name="1.2"></a>
+
+### 1.2 网络
+
+网络部分完成了网络的组网操作，，这一部分在[ppcls/arch/](../../../ppcls/arch/)下。 进入网络的数据将按照顺序(transforms->backbones->
```
Comment (TingquanGao) [inline docs/zh_CN/community/add_new_algorithm.md]: 1. 是transforms后的数据才会进入网络吧；
2. backbones -> backbone。

File: docs/zh_CN/community/add_new_algorithm.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+2. 在 my_backbone.py 文件内添加相关代码，示例代码如下:
+
+```python
+import paddle
+import paddle.nn as nn
+import paddle.nn.functional as F
+
+
+class MyBackbone(nn.Layer):
+    def __init__(self, *args, **kwargs):
```
Comment (TingquanGao) [inline docs/zh_CN/community/add_new_algorithm.md]: 这里是不是可以加上 class_num 参数，因为下面示例中用到了

File: docs/zh_CN/community/add_new_algorithm.md [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        pass
+
+    def forward(self, x, label):
+        loss = self.loss(input=predicts, label=label)
+        return {'your loss name': loss}
+```
+
+3. 在 [ppcls/loss/\__init\__.py](../../../ppcls/loss/__init__.py)文件内导入添加的模块。
+
+在损失函数添加之后，只需要配置文件中进行配置即可使用，如:
```
Comment (TingquanGao) [inline docs/zh_CN/community/add_new_algorithm.md:233]: 是否有必要说明一下loss在config中是list格式呢

File: docs/zh_CN/community/add_new_algorithm.md [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+
+    def get_metric(self, *args, **kwargs):
+        # you metric code
+        pass
+
+```
+
+3. 在 [ppcls/metric/\_\_init\_\_.py](../../../ppcls/metric/__init__.py)文件内导入添加的模块。
+
+在指标评估模块添加之后，只需要配置文件中进行配置即可使用，如:
```
Comment (TingquanGao) [inline docs/zh_CN/community/add_new_algorithm.md:273]: 是否有必要说明一下metric在config中是list格式呢，支持同时使用多种metric

File: docs/zh_CN/community/add_new_algorithm.md [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+<a name="1.6"></a>
+
+### 1.6 优化器
+
+优化器用于训练网络。优化器内部还包含了网络正则化和学习率衰减模块。 这一部分在[ppcls/optimizer/](../../ppcls/optimizer/)下。 PaddleClas内置了`Momentum`,`SGD`
+,`Adam`和`AdamW`等常用的优化器模块，`Constant`、`Linear`,`Cosine`,`Step`和`Piecewise`等常用的学习率衰减模块。
+
+对于没有内置的模块可通过如下步骤添加，以`optimizer`为例:
+
+1. 在 [ppcls/optimizer//optimizer.py](../../../ppcls/optimizer//optimizer.py) 文件内创建自己的优化器，示例代码如下:
```
Comment (TingquanGao) [inline docs/zh_CN/community/add_new_algorithm.md]: optimizer//optimizer.py -> optimizer/optimizer.py

File: docs/zh_CN/community/add_new_algorithm.md [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+
+```python
+from paddle import optimizer as optim
+
+
+class MyOptim(object):
+    def __init__(self, learning_rate=0.001, *args, **kwargs):
```
Comment (TingquanGao) [inline docs/zh_CN/community/add_new_algorithm.md:301]: 在优化器实现中，对于学习率配置的处理是否有必要说明呢？

File: docs/zh_CN/community/add_new_algorithm.md (hunk 4 above)
Comment (cuicheng01) [reply docs/zh_CN/community/add_new_algorithm.md:233]: 很多人对yaml格式不熟悉，感觉如果说list会绕晕

File: docs/zh_CN/community/add_new_algorithm.md (hunk 5 above)
Comment (cuicheng01) [reply docs/zh_CN/community/add_new_algorithm.md:273]: 同上

File: docs/zh_CN/community/add_new_algorithm.md (hunk 7 above)
Comment (cuicheng01) [reply docs/zh_CN/community/add_new_algorithm.md:301]: 这里应该不需要，要不然说的太细了，这里正常来说他也会参考其他优化器的实现

File: docs/zh_CN/community/add_new_algorithm.md (hunk 6 above)
Comment (cuicheng01) [reply docs/zh_CN/community/add_new_algorithm.md]: done

File: docs/zh_CN/community/add_new_algorithm.md (hunk 3 above)
Comment (cuicheng01) [reply docs/zh_CN/community/add_new_algorithm.md]: done

File: docs/zh_CN/community/add_new_algorithm.md (hunk 2 above)
Comment (cuicheng01) [reply docs/zh_CN/community/add_new_algorithm.md]: done

**Review by TingquanGao — APPROVED** [review]
